pkg/upgrade/migrate.go

```diff
@@ -160,27 +161,29 @@ func Migrate(ctxt context.Context, c client.Interface, nodename string) error {
 		return fmt.Errorf("failed to close host-local IPAM data dir directory: %s", err)
 	}
 
-	// Disable cni by setting datastoreReady to false.
+	// Disable cni by setting DatastoreReady to false.
 	log.Info("setting datastore readiness to false")
-	clusterInfo, err := c.ClusterInformation().Get(ctxt, "default", options.GetOptions{})
-	if err != nil {
-		return fmt.Errorf("failed to fetch cluster information: %s", err)
-	}
-	if *clusterInfo.Spec.DatastoreReady {
-		*clusterInfo.Spec.DatastoreReady = false
-		for i := uint(0); i < 5; i++ {
+	var clusterInfo *v3.ClusterInformation
+	for i := uint(0); i < 5; i++ {
+		clusterInfo, err = c.ClusterInformation().Get(ctxt, "default", options.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to fetch cluster information: %s", err)
+		}
+		if clusterInfo.Spec.DatastoreReady == nil || *clusterInfo.Spec.DatastoreReady {
```

Comment:
I'm unclear on why `DatastoreReady` was a `*bool` in the first place. As a plain bool, we wouldn't have had this nil problem.

Reply:
Yeah, same. I think it was to distinguish between "not set" which should mean "true", and explicitly set to "false", but in retrospect it probably could have just been a bool.